### DISCUSSION TITLE
Perbedaan scaler.fit_transform dan scaler.transform

### DISCUSSION THREAD

**Discussion by daryraa** [teamalgoritma/community#722] [Q&A] [answered]
Saya ingin bertanya terkait scaler.fit_transform dan scaler.transform, perbedaanya itu apa ya kenapa biasanya scaler.fit_transform itu buat data train sedangkan scaler.transform itu buat data pelatihan, kenapa menggunakan method yang berbeda?kalau misalnya scaler.fit_transform digunakan untuk data train dan data test itu bagaimana?
![image](https://github.com/teamalgoritma/community/assets/78478403/768662f9-691a-453b-84b4-8eea1e18d19b)

**Comment by arkanayudhistira** [comment, marked as answer]
Dear @daryraa,

`StandardScaler()` merupakan metode scaling yang memanfaatkan nilai mean (`μ`) and standar deviasi(`σ`). Fungsi `fit_transform()` merupakan gabungan dari 2 fungsi, yaitu:
- fit() : Fungsi untuk menghitung dan menyimpan nilai mean dan standar deviasi dari data
- transform() : Fungsi untuk melakukan scaling ke data berdasarkan nilai mean dan standar deviasi yang telah di fit(). 
Ketika fungsi `transform()` digunakan pada x_test, maka `StandardScaler()` akan mengambil data mean dan standar deviasi dari x_train (yang disimpan dari `fit_transform()`), dan melakukan scaling ke x_test berdasarkan nilai tersebut.

Alasan mengapa kedua proses tersebut dilakukan adalah untuk menyesuaikan range hasil scaling pada data test agar sesuai dengan range data train (merupakan range yang dipelajari oleh model). Jika `fit_transform()` digunakan pada data test juga, nanti hasil scalingnya akan berbeda dengan data train, menyebabkan perbedaan nilai yang dipelajari dan nilai yang akan diprediksi.

Semoga jawaban saya membantu.

**Reply by daryraa** [reply]
terimakasih
